python/feathub/feature_views/feature_view.py

```diff
@@ -192,3 +210,33 @@ def get_bounded_view(self) -> TableDescriptor:
         feature_view = deepcopy(self)
         feature_view.source = self.get_resolved_source().get_bounded_view()
         return feature_view
+
+    def _derive_feature_dtype(self, feature: Feature) -> Optional[DType]:
+        transform = feature.transform
+        if isinstance(transform, ExpressionTransform):
+            return self._derive_feature_dtype_from_feathub_expression(transform.expr)
+        elif isinstance(transform, OverWindowTransform) or isinstance(
+            transform, SlidingWindowTransform
+        ):
+            expr_result_type = self._derive_feature_dtype_from_feathub_expression(
+                transform.expr
+            )
+            return transform.agg_func.get_result_type(expr_result_type)
+        elif isinstance(transform, JoinTransform):
+            raise FeathubException("JoinTransform feature should have dtype set.")
+        else:
+            raise FeathubException(
+                f"Cannot derive feature data type of {type(transform)} feature"
+            )
+
+    def _derive_feature_dtype_from_feathub_expression(self, expr: str) -> DType:
+        ast = feathub_expr_parser.parse(expr)
+
+        variable_types = {
```

Comment:
Can we derive the `variable_types` once and re-use it to derive types for all Features in this FeatureView?